src/System.Collections.Immutable/tests/ImmutableInterlockedTests.cs

```diff
@@ -25,6 +25,18 @@ public void Update_StartWithNull()
             });
         }
 
+        [Fact]
+        public void UpdateArray_StartWithEmpty()
```

Comment:
How about a test for an *uninitialized* `ImmutableArray<T>` instance? (one where `array == null`)?

Reply:
Other APIs null ref in that case, as does this one. We want tests validating null refs?

Reply:
Sure. It's documented behavior. And we have [many other such NRE tests around ImmutableArray](https://github.com/dotnet/corefx/blob/95e35e10d7c52a7eea38c2ae6402d29e2a01efc6/src/System.Collections.Immutable/tests/ImmutableArrayTest.cs#L224).